### PR TITLE
refactor: Make span default ctr and assignment be `= default`

### PR DESCRIPTION
**Pull request by lgritz**
Minor simplification.

### CODE REVIEW

File: src/include/OpenImageIO/span.h [hunk 1]
```diff
@@ -143,11 +143,7 @@ class span {
         : span (il.begin(), il.size()) { }
 
     /// Assignment copies the pointer and length, not the data.
-    span& operator= (const span &copy) {
-        m_data = copy.data();
-        m_size = copy.size();
-        return *this;
-    }
+    constexpr span& operator= (const span &copy) = default;
```
Comment (ThiagoIze) [inline src/include/OpenImageIO/span.h:146]: This change is pretty close to what [std::span does](https://en.cppreference.com/w/cpp/container/span/operator%3D). The only bit missing, if you want it to fully match, is marking it noexcept:
```c++
constexpr span& operator=( const span& other ) noexcept = default;
```

**Review by ThiagoIze — APPROVED** [review]
Looks fine. You could mark it `noexcept` if it's important to you that you perfectly match std::span.

File: src/include/OpenImageIO/span.h (hunk 1 above)
Comment (lgritz) [reply src/include/OpenImageIO/span.h:146]: You're right. And I noticed a few other spots where we can be noexcept. Testing now and will amend the PR if it works.

Comment (lgritz) [reply src/include/OpenImageIO/span.h:146]: updated

**Review by ThiagoIze — APPROVED** [review]

**Pull request merged**